lib/next_ls.ex

```diff
@@ -592,12 +592,19 @@ defmodule NextLS do
             _ -> {name, GenLSP.Enumerations.CompletionItemKind.text(), ""}
           end
 
-        %GenLSP.Structures.CompletionItem{
-          label: label,
-          kind: kind,
-          insert_text: name,
-          documentation: docs
-        }
+        completion_item =
+          %GenLSP.Structures.CompletionItem{
+            label: label,
+            kind: kind,
+            insert_text: name,
+            documentation: docs
+          }
+
+        case NextLS.Snippet.get(name, nil) do
+          nil -> [completion_item]
+          snippets when is_list(snippets) -> [completion_item] ++ snippets
```

Comment:
I think the snippet should replace the completion item in most cases. 

Otherwise multiple items for let's say defmodule will be present.